File: 48-rotate-image/rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int m = matrix.size(), n = matrix[0].size();
        vector<vector<int>> result(m, vector<int>(n,0));
        for(int i = 0; i<m; i++){
            for(int j = 0; j<n; j++){
                result[j][n-1-i] = matrix[i][j];
            }
        }
        matrix = result;
    }
};

File: 48-rotate-image/rotate-image.py
class Solution(object):
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        m = len(matrix)
        n = len(matrix[0])
        for j in range(0,n):
            for i in range(0,(m+1)//2):
                matrix[i][j], matrix[m-i-1][j] =  matrix[m-i-1][j], matrix[i][j]
        
        #Transpose:
        for i in range(0, m):
            for j in range(i+1, n):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]